analytics/src/main/java/edu/artic/analytics/AnalyticsTracker.kt

```diff
@@ -29,10 +33,29 @@ interface AnalyticsTracker {
 class AnalyticsTrackerImpl(context: Context,
                            private val languageSelector: LanguageSelector,
                            private val membershipPrefs: MemberInfoPreferencesManager,
+                           private val locationService: LocationService,
                            private val analyticsConfig: AnalyticsConfig) : AnalyticsTracker {
 
     private val analytics = GoogleAnalytics.getInstance(context)
     private val tracker = analytics.newTracker(analyticsConfig.trackingId)
+    private var atMusuem = false
+    private var reportLocationAnalytic = true
+
+    init {
+        locationService.requestTrackingUserLocation()
+        locationService.currentUserLocation
```

Comment:
suggest change to 
`val isInMuseum = locationService.currentuserLocation.map{isLocationInMuseum(it)}`
then 
`isInMuseum.distinctUntilChanged().subscibeBy{ atMuseum = $it}`
`isInMuseum.filter{$it}.take(1).subscribe{ reportEvent(...)}`